helpers/csrf.js

```diff
@@ -12,7 +12,25 @@ const duplicateTokenHandler = (req, res, next) => {
 	next();
 };
 
+const errorHandler = (err, req, res, next) => {
+	if (err.code === 'EBADCSRFTOKEN') {
+		// convert body object to array
+		res.locals.csrfToken = req.csrfToken();
+		// send base URL for opening in new tab
+		const baseUrl = (req.headers.origin || process.env.HOST || 'http://localhost:3100');
```

Comment:
here use only origin, later when rendering and baseUrl or values or loggedin is not nefined, do not display the retry button but a load referer page (not browser back but re-load last page, with sadly loosing input data)